spec/requests/api/reports_spec.rb

```diff
@@ -0,0 +1,79 @@
+require "spec_helper"
+
+RSpec.describe "reports API" do
+  include Rack::Test::Methods
+
+  def app
+    Vmdb::Application
+  end
+
+  before { init_api_spec_env }
+
+  it "can fetch all the reports" do
+    report_1 = FactoryGirl.create(:miq_report_with_results)
+    report_2 = FactoryGirl.create(:miq_report_with_results)
+
+    api_basic_authorize
+    run_get "api/reports"
+
+    expect_result_resources_to_include_data(
+      "resources",
+      "href" => [
+        "http://example.org/api/reports/#{report_1.to_param}",
+        "http://example.org/api/reports/#{report_2.to_param}"
+      ]
+    )
+    expect_result_to_match_hash(@result, "count" => 2, "name" => "reports")
+    expect_request_success
+  end
+
+  it "can fetch a report" do
+    report = FactoryGirl.create(:miq_report_with_results)
+
+    api_basic_authorize
+    run_get "/api/reports/#{report.to_param}"
+
+    expect_result_to_match_hash(
+      @result,
+      "href"  => "http://example.org/api/reports/#{report.to_param}",
+      "id"    => report.id,
+      "name"  => report.name,
+      "title" => report.title
+    )
+    expect_request_success
+  end
+
+  it "can fetch a report's results" do
+    report = FactoryGirl.create(:miq_report_with_results)
+    result = report.miq_report_results.first
+
+    api_basic_authorize
+    run_get "/api/reports/#{report.to_param}/results"
+
+    expect_result_resources_to_include_data(
+      "resources",
+      "href" => [
+        "http://example.org/api/reports/#{report.to_param}/results/#{result.to_param}"
+      ]
+    )
+    expect_request_success
```

Comment:
none of the other tests need to specify http://example.org/ for matching href's, helper methods like expect_result_resources_to_include_hrefs("resources", href_list) would do matches on hrefs., look at vms_spec.rb:53 as an example. Thanks.